### PR TITLE
[Compute] sig image-version create: Support --os-snapshot and --data-snapshot

### PR DESCRIPTION
**Pull request by qwordy**
Issue #11117 

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR has modified HISTORY.rst describing any customer-facing, functional changes. Note that this does not include changes only to help content. (see [Modifying change log](https://github.com/Azure/azure-cli/tree/master/doc/authoring_command_modules#modify-change-log)).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

### CODE REVIEW

File: src/azure-cli/azure/cli/command_modules/vm/custom.py [hunk 1]
```diff
@@ -2768,19 +2768,31 @@ def create_gallery_image(cmd, resource_group_name, gallery_name, gallery_image_n
...
-    if not is_valid_resource_id(managed_image):
+    if managed_image and not is_valid_resource_id(managed_image):
         managed_image = resource_id(subscription=client.config.subscription_id, resource_group=resource_group_name,
                                     namespace='Microsoft.Compute', type='images', name=managed_image)
+    if os_snapshot and not is_valid_resource_id(os_snapshot):
+        os_snapshot = resource_id(subscription=client.config.subscription_id, resource_group=resource_group_name,
+                                  namespace='Microsoft.Compute', type='snapshots', name=os_snapshot)
+    data_snapshots = []
+    if data_snapshot is not None:
+        data_snapshots = data_snapshot.split()
```
Comment (MyronFanQiu) [inline src/azure-cli/azure/cli/command_modules/vm/custom.py]: you should use nargs='+'

File: src/azure-cli/azure/cli/command_modules/vm/_params.py [hunk 2]
```diff
@@ -793,6 +793,8 @@ def load_arguments(self, _):
                    help='Gallery image version in semantic version pattern. The allowed characters are digit and period. Digits must be within the range of a 32-bit integer, e.g. <MajorVersion>.<MinorVersion>.<Patch>')
         c.argument('description', help='the description of the gallery image version')
         c.argument('managed_image', help='image name(if in the same resource group) or resource id')
+        c.argument('os_snapshot', help='Name or ID of OS disk snapshot')
+        c.argument('data_snapshot', help='Names or IDs (space-delimited) of data disk snapshots')
```
Comment (MyronFanQiu) [inline src/azure-cli/azure/cli/command_modules/vm/_params.py]: please use nargs=+. you can find ref here https://docs.python.org/3/library/argparse.html#nargs

Comment (MyronFanQiu) [reply src/azure-cli/azure/cli/command_modules/vm/_params.py]: It cannot be updated. Right?

File: src/azure-cli/azure/cli/command_modules/vm/custom.py [hunk 3]
```diff
@@ -2768,19 +2768,31 @@ def create_gallery_image(cmd, resource_group_name, gallery_name, gallery_image_n
...
     end_of_life_date = fix_gallery_image_date_info(end_of_life_date)
-    if not is_valid_resource_id(managed_image):
+    if managed_image and not is_valid_resource_id(managed_image):
         managed_image = resource_id(subscription=client.config.subscription_id, resource_group=resource_group_name,
                                     namespace='Microsoft.Compute', type='images', name=managed_image)
+    if os_snapshot and not is_valid_resource_id(os_snapshot):
+        os_snapshot = resource_id(subscription=client.config.subscription_id, resource_group=resource_group_name,
+                                  namespace='Microsoft.Compute', type='snapshots', name=os_snapshot)
+    data_snapshots = []
+    if data_snapshot is not None:
```
Comment (MyronFanQiu) [inline src/azure-cli/azure/cli/command_modules/vm/custom.py]: This line always return true.

File: src/azure-cli/azure/cli/command_modules/vm/custom.py [hunk 4]
```diff
@@ -2789,13 +2801,26 @@ def create_image_version(cmd, resource_group_name, gallery_name, gallery_image_n
...
+            for i, s in enumerate(data_snapshots):
+                data_disk_images.append(GalleryDataDiskImage(source=GalleryArtifactVersionSource(id=s), lun=i))
+        if managed_image is None and os_snapshot is None:
+            raise CLIError('usage error: Please provide --managed-image or --os-snapshot')
+        storage_profile = GalleryImageVersionStorageProfile(source=source, os_disk_image=os_disk_image,
+                                                            data_disk_images=data_disk_images)
         image_version = ImageVersion(publishing_profile=profile, location=location, tags=(tags or {}),
                                      storage_profile=storage_profile)
     else:
+        if managed_image is None:
```
Comment (MyronFanQiu) [inline src/azure-cli/azure/cli/command_modules/vm/custom.py]: I cannot find the usage of `managed_image`. Why do you need to check it?

File: src/azure-cli/azure/cli/command_modules/vm/custom.py (hunk 3 above)
Comment (qwordy) [reply src/azure-cli/azure/cli/command_modules/vm/custom.py]: > 
> 
> This line always return true.

data_snapshots and data_snapshot  are different

File: src/azure-cli/azure/cli/command_modules/vm/custom.py (hunk 1 above)
Comment (qwordy) [reply src/azure-cli/azure/cli/command_modules/vm/custom.py]: > 
> 
> you should use nargs='+'

OK

File: src/azure-cli/azure/cli/command_modules/vm/_params.py (hunk 2 above)
Comment (qwordy) [reply src/azure-cli/azure/cli/command_modules/vm/_params.py]: > 
> 
> please use nargs=+. you can find ref here https://docs.python.org/3/library/argparse.html#nargs

OK

File: src/azure-cli/azure/cli/command_modules/vm/custom.py (hunk 4 above)
Comment (qwordy) [reply src/azure-cli/azure/cli/command_modules/vm/custom.py]: > 
> 
> I cannot find the usage of `managed_image`. Why do you need to check it?

It's used in profile.

File: src/azure-cli/azure/cli/command_modules/vm/_params.py (hunk 2 above)
Comment (qwordy) [reply src/azure-cli/azure/cli/command_modules/vm/_params.py]: > 
> 
> It cannot be updated. Right?

image-version cannot update the source (image or snapshot) now

**Pull request merged**